### PR TITLE
Jlou/feat terraform module

### PR DESCRIPTION
**Pull request by JoelLourenco**
# Azure Functions Deployment with Terraform

This Terraform module deploys Azure Functions with a dynamic packaging process. It creates the necessary Azure resources and uploads a ZIP package containing the function code.

## Module Structure

1. Terraform configuration files (`main.tf`, `variables.tf`, `outputs.tf`)

## Resources

- `random_string`: Generates a unique identifier for each deployment
- `azurerm_resource_group`: Creates or uses an existing resource group
- `azurerm_storage_account`: Creates a storage account for the function app
- `azurerm_service_plan`: Sets up the App Service Plan
- `null_resource`: Executes the packaging script
- `azurerm_linux_function_app`: Deploys the Azure Function App

## Deployment Process

1. **Resource Creation**:
   - A resource group is created if specified.
   - A storage account and service plan are set up.

2. **Package Creation**: 
   - A `null_resource` triggers the script to creates a ZIP file containing all function code and dependencies.

4. **Function App Configuration**:
   - The `azurerm_linux_function_app` resource is configured to run from the created package.
   - Environment variables are set in the app settings.

### CODE REVIEW

File: terraform/modules/azure_functions/main.tf [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  triggers = {
+    always_run = "${timestamp()}"
+  }
+
+  provisioner "local-exec" {
+    command = <<EOT
+      #!/bin/bash
+      set -e
+      set -x
+
```
Comment (vysahjk) [inline terraform/modules/azure_functions/main.tf:24]: on peut ajouter un check si la fonctionnalité "zip" est bien installé sur la machine ?

**Review by vysahjk — APPROVED** [review]
👍🏽

File: terraform/modules/azure_functions/main.tf (hunk 1 above)
Comment (JoelLourenco) [reply terraform/modules/azure_functions/main.tf:24]: C'est corrigé

**Review by vysahjk — COMMENTED** [review]
👍🏽

**Pull request merged**